web/src/billing/upgrade.ts

```diff
@@ -0,0 +1,104 @@
+import $ from "jquery";
+import {z} from "zod";
+
+import * as helpers from "./helpers";
+import {page_params} from "./page_params";
+
+type Prices = {
+    monthly: number;
+    annual: number;
+};
+
+type StripeSessionResponse = {
+    stripe_session_url: string;
+};
```

Comment:
We should move its corresponding schema into `helpers.ts` and use the inferred type of that schema, instead of manually rewriting `StripeSessionResponse`. Same thing in `event_status.ts`.
```
const response_schema = z.object({
    stripe_session_url: z.string(),
});
```